cloudinit/sources/DataSourceAzure.py

```diff
@@ -895,6 +896,18 @@ def _create_report_ready_marker(self):
             logger_func=LOG.debug,
         )
 
+    def _process_dhcp_lease(self, lease: dict) -> None:
+        """Maintain relevant info for leases."""
+        dhcp_opt_245 = lease.get("unknown-245")
+        if dhcp_opt_245:
+            self._wireserver_address = (
+                WALinuxAgentShim.get_ip_from_lease_value(dhcp_opt_245)
```

Comment:
move this function out of the shim? Previously its usage was restricted to the shim, if we're using it outside of the shim, we should just put it as a helper function in azure.py

Reply:
will do